File: Problems 1000-1099/1007. Minimum Domino Rotations For Equal Row/solution.cpp
class Solution {
    public:
        int minDominoRotations(vector<int>& tops, vector<int>& bottoms) {
            int n = tops.size();
            if (n == 0) return 0;
            // If solution exists, it will be either tops[0] or bottoms[0]
            vector<int> cand{ tops[0], bottoms[0] };
            for (int num : cand) {
                bool isCand = true;
                int topCount = 0;
                int bottomCount = 0;
                for (int i = 0; i < n; i++) {
                    if (tops[i] != num && bottoms[i] != num) {
                        isCand = false;
                        break;
                    }
    
                    if (tops[i] == num) topCount++;
                    if (bottoms[i] == num) bottomCount++;
                }
                if (isCand) {
                    return n - max(topCount, bottomCount);
                }
            }
            return -1;
        }
    };

File: Problems 1000-1099/1007. Minimum Domino Rotations For Equal Row/solution.py
class Solution:
    def minDominoRotations(self, tops: List[int], bottoms: List[int]) -> int:
        # Find if there is number that each domino has either top or
        # bottom (could be both if all dominoes have same two numbers)
        # Then minimum rotations would be the min dominoes that have
        # the number on top or bottom. For example, if 4 out of 6 dominoes
        # have 2 on top and other 2 dominoes have 2 on bottom, we rotate
        # the 2 dominoes that have 2 on bottom.

        n = len(tops)
        if n == 0: return 0
        
        # Numbers are from 1 to 6
        topCount = [0] * 7
        bottomCount = [0] * 7
        same = [0] * 7
        for i in range(n):
            topCount[tops[i]] += 1
            bottomCount[bottoms[i]] += 1
            if tops[i] == bottoms[i]:
                same[tops[i]] += 1

        for num in range(1, 7):
            if topCount[num] + bottomCount[num] - same[num] == n:
                return n - max(topCount[num], bottomCount[num])
        return -1
        
            